src/game/interpreter.rs

```diff
@@ -5,14 +5,80 @@ use crate::game::state;
 /// 
 /// # Arguments
 /// * `command` - A reference to a command from the ret_lang module.
+/// * `state` - A mutable reference to a GameState.
 ///
 /// # Returns
 /// * `Result<String, &str>` - A string that is the output of the command, or an error message.
-fn travel_interpreter<'a>(command: &'a ret_lang::Command) -> Result<String, &'a str> {
+fn travel_interpreter<'a>(command: &'a ret_lang::Command, state: &mut state::GameState) -> Result<String, &'a str> {
     match command {
         ret_lang::Command::Go(c) => {
-            let output = format!("Hero went {}", c.target);
-            Ok(output)
+            let (x, y) = match state.room {
+                Some(r) => r,
+                None => return Err("Not able to do that action right now."),
+            };
+            match c.target.to_lowercase().as_str() {
+                "north" => {
+                    let new_coords = (x, y - 1);
+                    let new_room = match state.map {
+                        Some(ref m) => m.get_room(new_coords.0, new_coords.1),
+                        None => return Err("Not able to do that action right now."),
+                    };
+                    let new_room = match new_room {
+                        Some(r) => {
+                            state.room = Some(new_coords);
+                            r
+                        },
+                        None => return Err("Not able to do that action right now."),
+                    };
+                    return Ok(format!("Hero went {}. {}", c.target, new_room.description));
```

Comment:
Refactor this repeated code so that it is more concise and easier to re-work to work with portals later.